models/Event.ts

```diff
@@ -29,19 +29,28 @@ export interface Event extends Sequelize.Model, ModelCommon<Event> {
   EventDetailId: number;
   EventDetail: DetailSnapShot;
   DeviceId: DeviceId;
+  Device: Device | null;
 }
+
+export interface QueryOptions {
+  eventType: string;
+  admin: boolean;
+  useCreatedDate: boolean;
+}
+
 export interface EventStatic extends ModelStaticCommon<Event> {
   query: (
     user: User,
     startTime: string,
```

Comment:
startTime, endTime, deviceId, offset and limit are all marked as optional at the express layer, but here the type signature assumes they're always defined.  You might be missing some handling of cases where they are actually undefined?  Change to i.e. `number | undefined` and see what edge cases you need to handle.